src/lib/Libtpp/tpp_router.c

```diff
@@ -2077,74 +2085,131 @@ router_pkt_handler(int tfd, void *data, int len, void *c, void *extra)
 						return 0;
 					}
 				} else if (orig_hop == 0) {
-					/* this to list of routers to whom we need to send */
-					if (!rlist) {
-						/* first element */
-						rsize = RLIST_INC;
-						rlist = malloc(sizeof(int) * rsize);
-						if (!rlist) {
-							if (cmprsd_len > 0)
-								free(minfo_base);
-							snprintf(tpp_get_logbuf(), TPP_LOGBUF_SZ, "Out of memory allocating pbs_comm list of %lu bytes",
-								(unsigned long)(sizeof(int) * rsize));
-							tpp_log_func(LOG_CRIT, __func__, tpp_get_logbuf());
-							if (data_out)
-								free(data_out);
-							return -1;
-						}
-						csize = 0;
-					}
-
+					/* add this to list of routers to whom we need to send */
 					/**
-					 * now check list backwards if router already sent to
+					 * now check list backwards if router already added to
```

Comment:
would saying
"now check list backwards if router is already added." make more sense?
this line and the next line do not seem to be part of one sentence.

Reply:
changed